### ISSUE TITLE
[Request] require auth on database deployment

### ISSUE THREAD

**Issue by wuast94** [lobehub/lobe-chat#4382]
### 🥰 Feature Description

as described in  #1406 it would be nice to force user auth BEFORE presenting the webapp interface. the initial FR got closed but it isnt implemented.

at the moment when a user opens the site not authentikated he sees the whole app. can enter in the chat and click on storage etc which cause just error toast messages in the top right.

he has to recognize that and have to click on the logo top left to then see the login feature. this is not user friendly and i know a lot of people that would never find or understand this. (Friends and family, dumb users etc) 

### 🧐 Proposed Solution

When not authenticated force a login screen

**Comment by cy948** [comment]
If you enable database serivce mode, that will require auth.

**Comment by wuast94** [comment]
no, it will display the normal webapp, you have to click on the lobechat icon top left and then click login/signup. all other thing will throw errors untlil loged in. 

if not logged in it should force you to sign in or up instead of displaying the normal webapp. its just not user friendly how it is at the moment.

**Comment by beliven-daniele-sarnari** [comment]
up 👍

**Comment by beliven-valentino-cossar** [comment]
+1 I also think this would be a valuable feature. Looking forward to seeing it implemented.

**Comment by mamachanko** [comment]
> Need a clever way to implement it. Maybe in the middleware part

@arvinxx usually frontend apps do what you just suggested. in the middleware requests are inspected for a token, when it is missing or expired, the user is redirected to the issuer.

+1 for this feature. if need be it could be feature flagged.

**Comment by arvinxx** [comment]
we have supported it

**Comment by beliven-valentino-cossar** [comment]
Hi @arvinxx, thanks for the update!
Is this the actual implementation of the feature? I wanted to confirm whether it’s possible to redirect users who are not logged in directly to the login page—ideally, without displaying the chat interface at all until they’re authenticated. Does the new approach handle this behaviour?

<img width="3439" height="1333" alt="Image" src="https://github.com/user-attachments/assets/18a4060b-6137-42c4-a4dd-5111ce65ef63" />

**Comment by beliven-valentino-cossar** [comment]
Is it implemented by the [environment variable `ENABLE_AUTH_PROTECTION`](https://lobehub.com/it/docs/self-hosting/environment-variables/basic#enable-auth-protection)?
